src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -154,101 +156,154 @@ protected EtlProcess(Transformation transformation, TConfiguration configuration
+                
+        private void ExtractCounters(DocumentsOperationContext context,
+            ExtractedItemsEnumerator<TExtracted> merged,
+            long fromEtag,
+            EtlStatsScope stats,
+            DisposableScope scope)
+        {
+            var lastDocEtag = stats.GetLastTransformedOrFilteredEtag(EtlItemType.Document);
 
-                        if (Transformation.ApplyToAllDocuments)
-                        {
-                            var counters = Database.DocumentsStorage.CountersStorage.GetCountersFrom(context, fromEtag, 0, long.MaxValue).GetEnumerator();
-                            scope.EnsureDispose(counters);
+            if (Transformation.ApplyToAllDocuments)
+            {
+                var counters = Database.DocumentsStorage.CountersStorage.GetCountersFrom(context, fromEtag, 0, long.MaxValue).GetEnumerator();
+                scope.EnsureDispose(counters);
 
-                            counters = new FilterCountersEnumerator(counters, stats, Database.DocumentsStorage, context, lastDocEtag);
+                counters = new FilterCountersEnumerator(counters, stats, Database.DocumentsStorage, context, lastDocEtag);
 
-                            merged.AddEnumerator(ConvertCountersEnumerator(counters, null));
+                merged.AddEnumerator(ConvertCountersEnumerator(counters, null));
+            }
+            else
+            {
+                foreach (var collection in Transformation.Collections)
+                {
+                    var counters = Database.DocumentsStorage.CountersStorage.GetCountersFrom(context, collection, fromEtag, 0, long.MaxValue).GetEnumerator();
+                    scope.EnsureDispose(counters);
 
-                        }
-                        else
-                        {
-                            foreach (var collection in Transformation.Collections)
-                            {
-                                var counters = Database.DocumentsStorage.CountersStorage.GetCountersFrom(context, collection, fromEtag, 0, long.MaxValue).GetEnumerator();
-                                scope.EnsureDispose(counters);
+                    counters = new FilterCountersEnumerator(counters, stats, Database.DocumentsStorage, context, lastDocEtag);
 
-                                counters = new FilterCountersEnumerator(counters, stats, Database.DocumentsStorage, context, lastDocEtag);
+                    merged.AddEnumerator(ConvertCountersEnumerator(counters, collection));
+                }
+            }
+        }
+        
+        private void ExtractTimeSeries(DocumentsOperationContext context,
+            ExtractedItemsEnumerator<TExtracted> merged,
+            long fromEtag,
+            EtlStatsScope stats,
+            DisposableScope scope)
+        {
+            var lastDocEtag = stats.GetLastTransformedOrFilteredEtag(EtlItemType.Document);
```

Comment:
What is we have 10000 time series append? What is the etag that would be recorded?
The document or the time series?

Reply:
I hope I follow you.
As I understand it we store the general last ETag (per node)
And we track for a batch in memory per type - document, counters, attachment, and now timeseries.
Why did you specify "10000"? Because it includes more than one time-series segment?